include/concurrencpp/executors/derivable_executor.h

```diff
@@ -31,10 +31,20 @@ namespace concurrencpp {
             return do_bulk_post(self(), callable_list);
         }
 
+        template<class callable_type>
+        void bulk_post(std::vector<callable_type> &callable_list) {
```

Comment:
Are we sure we need this addition? it seems like the changes in `executor.h` are suffice.

Reply:
The main library compiles without these workaround, yes. But the tests fail to compile, e.g.
```
[ 35%] Building CXX object CMakeFiles/inline_executor_tests.dir/source/tests/executor_tests/inline_executor_tests.cpp.o
/home/user/concurrencpp/test/source/tests/executor_tests/inline_executor_tests.cpp:208:15: error: no matching member function for call to 'bulk_post'
    executor->bulk_post<decltype(thrower)>(tasks);
    ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/concurrencpp/include/concurrencpp/executors/executor.h:115:14: note: candidate function template not viable: no known conversion from 'std::vector<decltype(thrower)>' (aka 'vector<(lambda at /home/user/concurrencpp/test/source/tests/executor_tests/inline_executor_tests.cpp:201:20)>') to 'std::span<(lambda at /home/user/concurrencpp/test/source/tests/executor_tests/inline_executor_tests.cpp:201:20)>' for 1st argument
        void bulk_post(std::span<callable_type> callable_list) {
```

Reply:
OK. 
I still worry about the `&` though, the overload will not work for `vector&&` for example (when returned from a function, for example `executor->bulk_post(produce_multiple_functors())` ).
we need to think about something more generic. 
If `tasks` are changed to be `std::array`, does it compile?